src/OpenTelemetry.Instrumentation.AspNetCore/AspNetCoreInstrumentationTracerProviderBuilderExtensions.cs

```diff
@@ -121,5 +121,12 @@ private static void AddAspNetCoreInstrumentationSources(
             builder.AddSource(HttpInListener.ActivitySourceName);
             builder.AddLegacySource(HttpInListener.ActivityOperationName); // for the activities created by AspNetCore
         }
+
+        // SignalR activities first added in .NET 9.0
+        if (Environment.Version.Major >= 9)
+        {
+            // https://github.com/dotnet/aspnetcore/blob/6ae3ea387b20f6497b82897d613e9b8a6e31d69c/src/SignalR/server/Core/src/Internal/SignalRServerActivitySource.cs#L13C35-L13C70
+            builder.AddSource("Microsoft.AspNetCore.SignalR.Server");
```

Comment:
are these long (as in, really long) running server spans? I recall ApplicationInsights users complained about recording such long running spans, as that'd skew the latency calculations, and ended up ignoring those spans by default... 

https://github.com/dotnet/aspnetcore/pull/32084

Reply:
If this is the case, then we should not do this by default, as there is no API to "remove" a source once added to the provider....

Reply:
The only long running span should be the http one that has the long running tag. All of the added SignalR spans are only for a specific operation.
![{4852D1D1-3D32-4C73-8B65-823B1B13C9B6}](https://github.com/user-attachments/assets/29eaacf9-c30c-4746-8c4e-166264885307)

Most of the spans added by SignalR add the long running span as a link, but individually they are short lived operations.
![{84468919-B48C-4E8C-9E4B-4793E39D5756}](https://github.com/user-attachments/assets/467e9522-f876-47e9-abdf-e3327c1eff53)
The `OnConnectedAsync` and `OnDisconnectedAsync` spans are the only 2 that are parented under the long running span, but they are both very short lived spans themselves.

Reply:
A few years back, I collaborated with your team to introduce the `http.long_running` attribute in SignalR to help filter long-running HTTP requests in Application Insights .NET SDK. This effort was driven by significant customer feedback about long-running SignalR requests distorting request duration metrics.

With `http.long_running` part of the default ASP.NET Core instrumentation, customers might face similar issues if long-running spans are automatically included without a suppression mechanism. I recommend considering an option to filter these spans by default or providing a toggle to enable them explicitly before rolling this out.